x-pack/plugins/cases/server/saved_object_types/migrations/get_all_persistable_attachment_migrations.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MigrateFunctionsObject } from '@kbn/kibana-utils-plugin/common';
+import { PersistableStateAttachmentTypeRegistry } from '../../attachment_framework/persistable_state_registry';
+import { PersistableStateAttachmentState } from '../../attachment_framework/types';
+
+const getMigrateFunction = (
+  persistableStateAttachmentTypeRegistry: PersistableStateAttachmentTypeRegistry
+) => {
+  const migrateFn = (state: PersistableStateAttachmentState, version: string) => {
+    let output = state;
+
+    if (!persistableStateAttachmentTypeRegistry.has(state.persistableStateAttachmentTypeId)) {
+      return output;
+    }
+
+    const attachment = persistableStateAttachmentTypeRegistry.get(
+      state.persistableStateAttachmentTypeId
+    );
+
+    const migrations = attachment.migrations;
+    const attachmentMigrations = typeof migrations === 'function' ? migrations() : migrations;
+
+    if (attachmentMigrations[version]) {
+      output = attachmentMigrations[version](state);
+    }
+
+    return output;
+  };
+
+  return migrateFn;
+};
+
+export const getAllPersistableAttachmentMigrations = (
```

Comment:
We need to construct a `MigrateFunctionsObject`. The key will be the Kibana version and for value a migration function that will run for all persistable state attachments of that version.